File: Data Structures & Algorithms/climbing-stairs/submission-2.cpp
class Solution {
public:
    int climbStairs(int n) {
        /*
        n=1, 1
        n=2, 2
        n=3, 3 n2+n1
        n=4, 1+1+1+1, 1+2+1 =>3, 2+2 5 n3+n2
        n=5, 1+1+1+1+1, 1+2+1+1 =>4, 2+2+1=>3 8 n4+n3
        */

        int dp[31];
        dp[0] = 0;
        dp[1] = 1;
        dp[2] = 2;
        
        for(int i=3; i<=30; i++)
        {
            dp[i] = dp[i-1]+dp[i-2];
        }
        return dp[n];
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int i=0, idxI = 0, idxJ = 0;
        while(i <nums.length) {
            int find = target - nums[i];
            idxI = i;
            Map<Integer, Integer> hm = new HashMap<>();
            for (int j = i+1; j < nums.length; j++) {
                hm.put(nums[j], j);
            }
            if (hm.get(find) != null) {
                idxJ = hm.get(find);
                break;
            }
            i++;
        }
        return new int[]{idxI, idxJ};
    }
}
